rmgpy/rmg/model.py

```diff
@@ -995,7 +1004,80 @@ def addSpeciesToEdge(self, spec):
         Add a species `spec` to the reaction model edge.
         """
         self.edge.species.append(spec)
-
+    
+    def setThermodynamicFilteringParameters(self,Tmax, toleranceThermoKeepSpeciesInEdge,minCoreSizeForPrune,maximumEdgeSpecies,reactionSystems):
+        """
+        sets parameters for thermodynamic filtering based on the current core
+        """
+        self.Tmax = Tmax
+        Gs = [spc.thermo.getFreeEnergy(Tmax) for spc in self.core.species]
+        self.Gmax = max(Gs)
+        self.Gmin = min(Gs)
+        
+        self.Gfmax = toleranceThermoKeepSpeciesInEdge*(self.Gmax-self.Gmin)+self.Gmax
+        self.toleranceThermoKeepSpeciesInEdge = toleranceThermoKeepSpeciesInEdge
+        self.minCoreSizeForPrune = minCoreSizeForPrune
+        self.reactionSystems = reactionSystems
+        self.maximumEdgeSpecies = maximumEdgeSpecies
+        
+    def thermoFilterDown(self,maximumEdgeSpecies,minSpeciesExistIterationsForPrune=0):
```

Comment:
I like the description filtering better (since pruning seems like gardening jargon), though I feel that since this is similar to the function of what we already called pruning, we could keep it consistent. maybe have `thermoPrune` and `fluxPrune` or `kineticsPrune` for the other method.